core/src/main/java/com/xatkit/core/recognition/nlpjs/NlpjsIntentRecognitionProvider.java

```diff
@@ -0,0 +1,210 @@
+        this.intentsToRegister = new HashMap<>();
+        this.entitiesToRegister = new HashMap<>();
+    }
+
+    @Override
+    public void registerEntityDefinition(@NonNull EntityDefinition entityDefinition) throws IntentRecognitionProviderException {
+        checkNotShutdown();
+        if (entityDefinition instanceof BaseEntityDefinition) {
+            BaseEntityDefinition baseEntityDefinition = (BaseEntityDefinition) entityDefinition;
+            if (nlpjsEntityReferenceMapper.isSupported(baseEntityDefinition.getEntityType())) {
+                throw new IntentRecognitionProviderException(MessageFormat.format("Cannot register the entity " +
+                        "{0}. This type is not supported by NLP.js", entityDefinition));
+            }
+        } else if (entityDefinition instanceof CustomEntityDefinition) {
+            Log.debug("Registering {0} {1}", CustomEntityDefinition.class.getSimpleName(), entityDefinition.getName());
+            if (entityDefinition instanceof CompositeEntityDefinition) {
+                throw new IntentRecognitionProviderException(MessageFormat.format("Cannot register the entity " +
+                        "{0}. Composite entities are not supported by NLP.js", entityDefinition));
+            } else {
+                Entity entity = this.nlpjsEntityMapper.mapEntiyDefinition(entityDefinition);
+                this.entitiesToRegister.put(entityDefinition.getName(), entity);
+            }
+
+        } else {
+            throw new IntentRecognitionProviderException(MessageFormat.format("Cannot register the provided {0}, " +
+                            "unsupported {1}", entityDefinition.getClass().getSimpleName(),
+                    EntityDefinition.class.getSimpleName()));
+        }
+    }
+
+    @Override
+    public void registerIntentDefinition(@NonNull IntentDefinition intentDefinition) throws IntentRecognitionProviderException {
+        checkNotShutdown();
+        checkNotNull(intentDefinition.getName(), "Cannot register the %s with the provided name %s",
+                IntentDefinition.class.getSimpleName(), intentDefinition.getName());
+        if (this.intentsToRegister.containsKey(intentDefinition.getName())) {
+            throw new IntentRecognitionProviderException(MessageFormat.format("Cannot register the intent {0}, the " +
+                    "intent already exists", intentDefinition.getName()));
+        }
+        Log.debug("Registering NLP.js intent {0}", intentDefinition.getName());
+        Intent intent = nlpjsIntentMapper.mapIntentDefinition(intentDefinition);
+        this.intentsToRegister.put(intentDefinition.getName(), intent);
+    }
+
+    @Override
+    public void deleteEntityDefinition(@NonNull EntityDefinition entityDefinition) throws IntentRecognitionProviderException {
+
+    }
+
+    @Override
+    public void deleteIntentDefinition(@NonNull IntentDefinition intentDefinition) throws IntentRecognitionProviderException {
+
```

Comment:
add `throw new UnsupportedOperationException("Not implemented");`
I'll take care of this.